C-Sharp.md

```diff
@@ -1,6 +1,232 @@
 #The C# Programming Language
 
+1. ### Basics
+  * Setup
+  * Your first C# program
+  * Types and variables
+  * Flow control statements
+  * Strings
+  * Classes, Objects, Interface and methods
+  * Fields and properties
+  * Scope and accessibility modifiers
+  * Handling exceptions
 
-* Introduction
-* Setup
-* Language features 
+2. ### Intermediate
+  * Generics
+  * Events, delegates and lambda expressions
+  * Collection framework
+  * LINQ
+
+3. ### Advanced
+  * Asynchronous Programming (Async and Await)  
+  * Task Parallel Library
+
+5. ### What's New in C# 6
+  * Null-Conditional operator
+  * Auto-Property initializers
+  * Nameof expressions
+  * Expression bodied functions and properties
+  * Other features
+
+4. ### Object-Oriented principles (OOP)
+   * Encapsulation
+   * Abstraction
+   * Inheritance
+   * Polymorphism
+
+6. ### Solid principles
+  * Single responsibility principle
+  * Open closed principle
+  * Liskov substitution principle
+  * Interface segregation principle
+  * Dependency inversion principle
+
+7. ### C# Best practices, Design patterns & Test driven development (TDD)
+
+###Setup
+[LinqPad][linqpad] is an ultimate .NET scratchpad to quickly test your C# code snippets. Standard edition is free and perfect tool for beginners to quickly execute language statements, expressions and programs.
+
+Alternatively, you could also download [Visual Studio Community 2015]  [visualstudio]  which is an extensible [IDE][ide] used by most professionals for creating enterprise applications.
+
+###Your First C# Program
+
+```csharp
+//this is the single line comment
+
+/** This is multiline comment,
+compiler ignores any code inside comment blocks.
+**/
+
+//This is the namespace, part of the standard .NET Framework Class Library
+using System;
+// namespace defines the scope of related objects into packages
+namespace Learning.CSharp
+{  
+  // name of the class, should be same as of .cs file
+  public class Program
+  {
+    //entry point method for console applications
+   public static void Main()
+    {
+      //print lines on console
+      Console.WriteLine("Hello, World!");
+      //Reads the next line of characters from the standard input stream.Most common use is to pause program execution before clearing the console.
+      Console.ReadLine();
+    }
+  }
+}
+```
+If you are using visual Studio, follow [this](https://msdn.microsoft.com/en-us/library/k1sx6ed2.aspx) to create console application and understand your first C# program.
+
+###Types and variables
+C# is a strongly typed language. Every variable has a type.
```

Comment:
"Strongly Typed" is going to need far more explanation, I think.  This is a really important topic and a large departure from most other interpreted languages.

It may be worth talking about how things are stored in memory, what their representation is, and what it means to have a datatype, from the compiler's perspective.